Add user task scheduling docs section

koevskinikola opened the pull request:
## What is the purpose of the change

With https://github.com/camunda/product-hub/issues/215, Camunda Platform 8 supports user task scheduling with the `dueDate` and `followUpDate` attributes.

## Are there related marketing activities

/

## When should this change go live?

With version `8.2.0`.

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

## Related issue

closes #1733

koevskinikola commented on docs/components/modeler/bpmn/user-tasks/user-tasks.md:
```diff
@@ -55,6 +55,29 @@ The unique identifier depends on the authentication method used to login to Task
+expressions are evaluated on activating the user task and must result in a `string` conforming to an ISO 8601 combined
+date and time representation.
+
+:::note
+An ISO 8601 combined date and time representation must contain timezone information, either `Z` for UTC or a zone offset.
+Optionally, it can contain a zone id. The following examples represent valid ISO 8601 `strings`:
+
+- `2019-10-01T12:00:00Z` - UTC time
+- `2019-10-02T08:09:40+02:00` - UTC plus two hours zone offset
+- `2019-10-02T08:09:40+02:00[Europe/Berlin]` - UTC plus two hours zone offset at Berlin
+  :::
```
This paragraph is duplicated from the [Timer docs](https://docs.camunda.io/docs/next/components/modeler/bpmn/timer-events/#time-date). Might be good to get some hints from a technical writer.

christinaausley replied:
I don't have any technical adjustments for this commit -- but taking a closer look at this, are you worried this section may not be as understandable to the user?

koevskinikola replied:
Hey @christinaausley, I'm worried about the duplication of the content and future maintainability. I think it should be pretty clear to the user.

Are there any guidelines on how to avoid duplicated content for the docs? I couldn't find any.

christinaausley replied:
Ah, this is a great question.

@akeller and I had this discussion some time ago when we found ourselves copying over instructions to create a cluster. For maintainability purposes, I ended up creating a React component for this we could call anytime:

<img width="1312" alt="Screen Shot 2023-03-10 at 9 25 44 AM" src="https://user-images.githubusercontent.com/84338309/224369478-cee99244-2c73-45b1-b31c-8e1674d54042.png">

I find this useful when there are several documents which need to refer to the same concept.

koevskinikola replied:
@christinaausley that's great! I used it for the ISO 8601 format. Let me know if it's all good.

koevskinikola commented on docs/components/modeler/bpmn/user-tasks/user-tasks.md:
```diff
@@ -31,7 +31,7 @@ extension element of the process element.
 
 User tasks support specifying assignments, using the `zeebe:AssignmentDefinition` extension element.
-This can be used to define which user the task can be assigned to. One or both of the following
+This can be used to define which user the task can be assigned to. One or all of the following
```
I piggybacked this small fix here, as it has a very small impact.

koevskinikola commented:
While https://github.com/camunda/product-hub/issues/215 hasn't been scheduled for `8.2.0-alpha5`, this PR can be included with the alpha release since it only documents what is supported through Zeebe.

koevskinikola commented:
UPDATE: ZPA is not targeting the `8.2.0-alpha5` release as some of the Zeebe workflow engine PRs related to this feature haven't been merged yet.

berkaycanbc commented on docs/components/modeler/bpmn/user-tasks/user-tasks.md:
```diff
@@ -55,6 +55,29 @@ The unique identifier depends on the authentication method used to login to Task
+expressions are evaluated on activating the user task and must result in a `string` conforming to an ISO 8601 combined
+date and time representation.
+
+:::note
```
:wrench: What about adding an example use case as we did in assignments section? Would be quite helpful to clearly understand the new feature.

koevskinikola replied:
That's a nice idea. Thanks @berkaycanbc.

Since this PR focuses only on the Zeebe part, I will add an example use-case on how users can use these dates with Zeebe, independently of Tasklist.

berkaycanbc reviewed (changes requested):
Clear explanation of the due and follow up dates feature. :rocket: 

I only requested a minor change about adding a use case of the feature.

akeller commented:
> UPDATE: ZPA is not targeting the `8.2.0-alpha5` release as some of the Zeebe workflow engine PRs related to this feature haven't been merged yet.

Should we apply the `hold` label then? Is it likely to make it to the 8.2 minor release?

koevskinikola commented:
@akeller this feature must make the `8.2.0` minor release as it was promised to a customer for that release. :slightly_smiling_face: 

In any case, the last code PR is reviewed right now, so I expect the Zeebe side will be wrapped in the next 7 days.

koevskinikola commented:
@berkaycanbc I've implemented your review hints. 

I added a more general use-case example that doesn't mention any Camunda Platform 8 applications since this PR is related to the Zeebe functionality of this feature.

I also expanded the XML code snipped so users will have an idea of how task scheduling will look in their BPMN process.

koevskinikola commented:
@berkaycanbc I refactored the note on ISO 8601 dates after your review. No technical documentation was changed, just how it's formatted.

christinaausley reviewed (approved):
No adjustments on this one 👍 However, do you want this to apply to the current version of docs rather than just the next? If so, we'll need to apply to the `8.1` docs.

koevskinikola commented:
@christinaausley I don't think we need to do this in the `8.1` docs. We won't backport the "User task due and follow-up dates" feature, so there won't be any duplication in the docs.

berkaycanbc reviewed (approved):
@koevskinikola Great updates to the documentation. Thanks for giving attention to these! :rocket:
